### PR TITLE
Refactor the indexing part of the `mini-blocks`

### PR DESCRIPTION
**Pull request by miiu96**
- Refactored the indexing part of the `mini-blocks`. 
- The old implementation had to do GET requests in order to know how to index the `mini-blocks`, this behavior generate some problems because there are conflicts. (the indexer had to do retries)  In order to fix the problem now mini-blocks are indexed with a `scripted_upsert` operation that will fix all these problems.

### CODE REVIEW

File: process/elasticproc/miniblocks/serialize.go [hunk 1]
```diff
@@ -32,28 +31,40 @@ func (mp *miniblocksProcessor) SerializeBulkMiniBlocks(
...
 	}
 
 	// prepare data for update operation
 	meta := []byte(fmt.Sprintf(`{ "update" : {"_index":"%s", "_id" : "%s" } }%s`, index, converters.JsonEscape(mbHash), "\n"))
-	if shardID == miniblockDB.SenderShardID && miniblockDB.ProcessingTypeOnDestination != block.Processed.String() {
-		// prepare for update sender block hash
-		serializedData := []byte(fmt.Sprintf(`{ "doc" : { "senderBlockHash" : "%s", "procTypeS": "%s" } }`, converters.JsonEscape(miniblockDB.SenderBlockHash), converters.JsonEscape(miniblockDB.ProcessingTypeOnSource)))
 
-		return meta, serializedData, nil
+	onSourceNotProcessed := shardID == miniblockDB.SenderShardID && miniblockDB.ProcessingTypeOnDestination != block.Processed.String()
```
Comment (bogdan-rosianu) [inline process/elasticproc/miniblocks/serialize.go:46]: notProcessedOnSource

File: process/elasticproc/miniblocks/serialize.go [hunk 2]
```diff
@@ -32,28 +31,40 @@ func (mp *miniblocksProcessor) SerializeBulkMiniBlocks(
...
+		}
 	}
+`
 
-	// prepare for update receiver block hash
-	serializedData := []byte(fmt.Sprintf(`{ "doc" : { "receiverBlockHash" : "%s", "procTypeD": "%s" } }`, converters.JsonEscape(miniblockDB.ReceiverBlockHash), converters.JsonEscape(miniblockDB.ProcessingTypeOnDestination)))
+	serializedDataStr := fmt.Sprintf(`{"scripted_upsert": true, "script": {`+
+		`"source": "%s",`+
+		`"lang": "painless",`+
+		`"params": { "mb": %s, "osnp": %t }},`+
```
Comment (bogdan-rosianu) [inline process/elasticproc/miniblocks/serialize.go:64]: osnp -> npos

File: integrationtests/miniblocks_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	"testing"
+
+	dataBlock "github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	indexerdata "github.com/multiversx/mx-chain-es-indexer-go/process/dataindexer"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIndexMiniBlocksOnSourceAndDestination(t *testing.T) {
+	setLogLevelDebug()
```
Comment (sstanculeanu) [inline integrationtests/miniblocks_test.go:16]: why is this needed?

File: process/elasticproc/miniblocks/serialize.go (hunk 1 above)
Comment (miiu96) [reply process/elasticproc/miniblocks/serialize.go:46]: done

File: process/elasticproc/miniblocks/serialize.go (hunk 2 above)
Comment (miiu96) [reply process/elasticproc/miniblocks/serialize.go:64]: done

File: integrationtests/miniblocks_test.go (hunk 3 above)
Comment (miiu96) [reply integrationtests/miniblocks_test.go:16]: just for an informative purpose, to see logs with Debug when integration tests are running

**Review by bogdan-rosianu — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]
